test/object-store/frozen_objects.cpp

```diff
@@ -289,6 +338,50 @@ TEST_CASE("Freeze Results", "[freeze_results]") {
         realm->close();
         REQUIRE(DB::call_with_lock(config.path, [](auto) {}));
     }
+
+    SECTION("Results after source remove") {
+        Results results;
+        std::function<void()> do_remove;
+
+        SECTION("Results on collection") {
+            Obj obj;
+            write([&]() {
+                obj = create_object(42, true);
+            });
+
+            SECTION("Dictionary") {
+                results = Results(realm, obj.get_dictionary_ptr(int_dict_col));
+            }
+
+            SECTION("Links") {
+                results = Results(realm, obj.get_linklist_ptr(object_link_col));
+            }
+
+            do_remove = [&, key = obj.get_key()] {
+                table->remove_object(key);
+            };
+        }
+
+        SECTION("Results on table") {
+            results = Results(realm, table);
+            do_remove = [&] {
+                realm->read_group().remove_table(table->get_key());
+            };
+        }
+
+        // FIXME? the test itself passes but crashes on teardown in notifier thread
+        //       with realm::NoSuchTable on Query constructor through import_copy_of
```

Comment:
Is it something that worth fixing also in some other pr?

Reply:
I'm just not sure it's valid usecase after looking at crashing usecases. Could you tell if this looks relatively sane thing to expect from usage of this api?